### DISCUSSION TITLE
How to remove ability to deselect value in Select component

### DISCUSSION THREAD

**Discussion by sabrinabrogren** [mantinedev/mantine#5917] [Q&A] [answered]
I'm not sure if this is a bug or I'm just using the Select in an unintended way. I don't want users to be able to deselect a value in the Select component. I have a controlled Select component where the `onChange` just doesn't do anything if the value is `null` (when an element is deselected). However, this still clears the Selected value until the user clicks elsewhere on the page. Is this a bug, or should I make my own Combobox element for this purpose?

Here is a recording of the behavior:
https://github.com/mantinedev/mantine/assets/15618941/d9a556b8-dcac-4fff-b9c2-9999c6acf689

[And here is a minimum reproduction in codesandbox](https://codesandbox.io/p/sandbox/old-rain-sfj65v?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cltpr304100073b6eqpox4ppc%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cltpr304000033b6ejg0flczo%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cltpr304000043b6edfzw14mr%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cltpr304000063b6edwj3gxwy%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cltpr304000033b6ejg0flczo%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cltpr304000023b6e6800u8wd%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522cltpr304000033b6ejg0flczo%2522%252C%2522activeTabId%2522%253A%2522cltpr304000023b6e6800u8wd%2522%257D%252C%2522cltpr304000063b6edwj3gxwy%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cltpr304000053b6esps8rowb%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522cltpr304000063b6edwj3gxwy%2522%252C%2522activeTabId%2522%253A%2522cltpr304000053b6esps8rowb%2522%257D%252C%2522cltpr304000043b6edfzw14mr%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522cltpr304000043b6edfzw14mr%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by rtivital** [comment, marked as answer]
`<Select allowDeselect={false} />`

**Reply by sabrinabrogren** [reply]
Oh wow, can't believe I skimmed right by that when reading through the docs. Thank you! 🙏
